homeassistant/components/sleepiq/entity.py

```diff
@@ -26,21 +40,21 @@ def __init__(
         sleeper: SleepIQSleeper,
         name: str,
     ) -> None:
-        """Initialize the SleepIQ side entity."""
+        """Initialize the SleepIQ sensor entity."""
         super().__init__(coordinator)
-        self.bed = bed
         self.sleeper = sleeper
-        self._async_update_attrs()
-
-        self._attr_name = f"SleepNumber {bed.name} {sleeper.name} {SENSOR_TYPES[name]}"
-        self._attr_unique_id = f"{bed.id}_{sleeper.name}_{name}"
+        self.bed = bed
         self._attr_device_info = DeviceInfo(
```

Comment:
You could make this smaller in the future by making a function to generate the device info from the bed object